FreeRTOS/FreeRTOS-Community-Supported-Demos
# Adding updates to f2838x demos

Adding updates to f2838x demos

Description
-----------
Following changes are added in this commit:

- Added F2838x example to showcase FreeRTOS application on both CPU1 and CPU2 cores.
- Added F2838x example to showcase interrupt latency in FreeRTOS context through ERAD IP.
- Updated FreeRTOS root path in projectspec as per Third Party folder structure.
- Added COFF related changes in Regtests
- Updated linker cmd file to create a separate memory section for FreeRTOS specific stack and heap
- Added freertos_config folder to keep project specifc FreeRTOS_config.h file

Test Steps
-----------
To run the demos on TMDSCNCD28388D controlcard, the following steps are required:
 - Install Code Compser Studio(CCS): https://www.ti.com/tool/CCSTUDIO
 - Install latest C2000Ware package: https://www.ti.com/tool/C2000WARE
 - Update the C2000WARE_ROOT and FreeRTOS_ROOT path variable in the example projectspec files available at following location in the repo: FreeRTOS-Community-Supported-Demos\C2000_F2838x_C28x_CCS\CCS
 - Import the example project file in CCS
 - Build and debug the project


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
